### DISCUSSION TITLE
"Download Failed" for Last Month on Chrome

### DISCUSSION THREAD

**Discussion by GH5012** [aclap-dev/video-downloadhelper#960] [Help and Q&A] [answered]
### Your question

I've been using this extension for years now. Roughly a month ago something changed and now my extension for Chrome won't work on any website i've attempted, including websites that this extension has worked on for years. Every video I attempt to download gives the same message:

"Download failed

Sadly we failed at downloading that specific media. We try to support as many websites as possible, so it would help a lot if you could report that error (it's anonymous!)."

I've tried clearing history/cache and restarting. I've attempted reinstalls of both the extension and companion app with no change. Companion app license is valid and re-entered after reinstall. I have available disk space.

Version 9.0.2.12 (stable)

Companion App installed (v2.0.19) 

Chrome Version 126.0.6478.127 

Windows 11 Home

### Addon & CoApp version and details

Version 9.0.2.12 (stable)

Companion App installed (v2.0.19) 

Chrome Version 126.0.6478.127 

Windows 11 Home

**Comment by jcfrog** [comment]
If nothing works, it's probably a network or blocker issue. Firewall, antivirus, proxy...

**Reply by GH5012** [reply]
Thanks for the reply. I do use Private Internet Access and Bitdefender Total Security that i've used for years as well alongside this extension, if you happen to be any aware of any new issues with those.

One reason that makes me suspicious that it's extension or companion app update related, is if you search "Download Failed" into these forums you get about 5 pages of results. About 3 of those 5 pages of search results have post dates between May 23rd-28th which is exactly when my problem started.

**Reply by jcfrog** [reply]
It's probably the time this feature has been released (the "Download failed" msg + report button). The amount of video sites is huge, we have a lot of reports, we can't pretend to work on every sites. But a lot are working fine. There must be a configuration issue if nothing works.

**Reply by GH5012** [reply, marked as answer]
It appears it was Bitdefender. Under my Bitdefender Firewall rules "net.downloadhelper.coapp-win-64.exe" was allowed, but "vdhcoapp.exe" was not. I allowed it and attempted a video download, still wouldn't work, cleared cache/history and restarted and still wouldn't work.

Dug a little bit more and saw that "ffprobe.exe" and "ffmpeg.exe" were both associated with the coapp. Both were listed as not allowed under firewall rules. Enabled both and video downloads would actually start now, but would get stuck at around 1% and stop. Cleared cache/history and restarted and now downloads from two different websites just worked.

So if anyone else reads this with a similar problem and you have Bitdefender, make sure all 3 of those things are allowed under your Bitdefender Firewall rules. I don't remember ever having to do that in the past, but apparently something with the new version got flagged.

**Reply by jcfrog** [reply]
thanks a lot for the feedback, it will help!

**Reply by GrampaWildWilly** [reply]
The new thing is the CoApp.  Past versions of the CoApp, versions 1.x, were named `net.downloadhelper.coapp-win-64.exe`, a fact you apparently knew in the past since that's what was allowed through your firewall.  But with CoApp 2.x, the file names changed.  Names.  Plural.  Used to be only one file.  Now there's a few of them.  Even the directory name changed.  This was to allow people to keep both versions of the CoApp in case they wanted to switch back for some reason.  I have never found a reason but then maybe I don't have a vivid enough imagination.  So you must have upgraded to the new CoApp & that's when things started breaking down for you.  At least, that's my best guess.

But rather than identifying individual .exe files as exceptions in the firewall, you should just mark the whole CoApp directory as an exception.  Files have been added to that directory over time.  You don't know what some future incarnation of the CoApp might require.  Just make the whole directory an exception to the firewall.  For good measure, do the same with the antivirus.

**Reply by GH5012** [reply]
I won't claim to be the most proficient with this, but it doesn't appear that I can create a rule for the entire CoApp directory. In Bitdefender under the Rules section of Firewall, it has a pre-populated list of programs that included net.downloadhelper.coapp-win-64.exe, vdhcoapp.exe, ffprobe.exe, and ffmpeg.exe. All 4 show up as individual programs. The last two show up differently than the first two, and I have to click to expand them on the list to see the option to enable them. Searching for "downloadhelper" on the rules list does bring up all 4 programs.

I do have the option to manually create a rule in Bitdefender under the Firewall Rules section where I can pick from my File Explorer. It does not appear that I can create a rule for the entire "DownloadHelper CoApp" file if that's what you're suggesting. It brings up an "Invalid application path error". It appears, at least from what I can see, that I can only create rules for individual .exe files.

**Reply by GrampaWildWilly** [reply]
Oh.  OK.  I didn't know that, seeing as how I don't use Bitdefender & never have.  Well, in that case, that's too bad.  It would certainly simplify things if you could do that.  But you can't.  Oh well.  It seems you have done what you can.  If things are working now, this must be the best you can do.
